### PR TITLE
Add instantiation-time support of protocol headers to sync thrift client.

### PR DESCRIPTION
**Pull request by junchaowu**

### CODE REVIEW

File: tchannel/sync/thrift.py [hunk 1]
```diff
@@ -70,11 +70,15 @@ def client_for(service, service_module, thrift_service_name=None):
...
-    def init(self, tchannel_sync, hostport=None, trace=False):
+    def init(self, tchannel_sync,
+             hostport=None,
+             trace=False,
+             protocol_headers=None):
         self.async_thrift = self.__async_client_class__(
             tchannel_sync._async_client,
             hostport,
             trace,
+            protocol_headers,
```
Comment (abhinav) [inline tchannel/sync/thrift.py]: Can you change this to use keyword args for all arguments?

The number of positional args here is getting unwieldy.

**Comment by abhinav** [comment]
Looks good besides minor styling change.

File: tchannel/sync/thrift.py (hunk 1 above)
Comment (breerly) [inline tchannel/sync/thrift.py]: What are all the args are being overloaded into `protocol_headers`? I just assumed it was a free-form dict...

**Comment by abhinav** [comment]
LGTM. Address @breerly's concerns and :shipit:.

File: tchannel/sync/thrift.py (hunk 1 above)
Comment (junchaowu) [inline tchannel/sync/thrift.py]: Yes. it is a dict.

**Pull request merged**